common/core/web/keyboard-processor/src/text/keyMapping.ts

```diff
@@ -18,6 +18,7 @@ namespace com.keyman {
       //ffie['k107'] = 187; // =    // FF 3.0 // I2062
       this.FF['k61'] = 187;  // =   // FF 2.0
       this.FF['k59'] = 186;  // ;
+      this.FF['k173'] = 189; // -/_
```

Comment:
This is a new mapping. Can you describe how you found that this was needed, and which versions of Firefox it has been needed for? Is this true for FF on all platforms?

Are there any other mappings we are missing? Also given those other commented entries from 2012, might be a good opportunity to clean them out.

Reply:
> This is a new mapping. Can you describe how you found that this was needed

Uh... I read the issue?  Noting that it was a FF-specific thing, my mind _very_ quickly went to the FF-specific keymapping section.  Then it was a matter of determining it was the `_` not being mapped, then confirming that the keycode didn't match Chrome's keycode.

For your other questions... they'll take additional time and effort to answer.  I'm fairly certain "true for FF on all platforms" == true, but the others... :shrug:

Reply:
https://css-tricks.com/snippets/javascript/javascript-keycodes/ strongly suggests it's just these three.

https://www.freecodecamp.org/news/javascript-keycode-list-keypress-event-key-codes/ reports a few others that are irrelevant, as they're media control keys.

I haven't found any references that indicate any form of platform-specific behavior for these three keys.  The closest thing is a Mac-specific behavior... for other keys... for the `keypress` event only.  (Check the first link.)

Finally found the old MDN table, which could _probably_ use an update:  https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

That said, note the details about Gecko (the Firefox engine).  A Google search shows that version 29 was released April 29th, 2014, so this has been around for a while.

Reply:
The comments suggest Firefox 2.0 ... with some discrepancies fixed in 3.0. That's a long time ago. Like 2005 or so.

https://github.com/keymanapp/keyman/blob/v2.0.377/source/kmwkeymaps.js#L26-L31